galata/test/jupyterlab/kernel.test.ts

```diff
@@ -104,6 +104,82 @@ test.describe('Kernel', () => {
         'Python 3 (ipykernel)'
       );
     });
+
+    test('Should support opening subshell in separate code console ', async ({
+      page
+    }) => {
+      // Open new notebook
+      await page.menu.clickMenuItem('File>New>Notebook');
+      await page
+        .getByRole('tab', { name: 'Untitled.ipynb' })
+        .waitFor({ state: 'detached' });
+      await page.getByRole('button', { name: 'Select Kernel' }).click();
+
+      // Run %subshell in notebook
+      const notebook = page.locator('.jp-Notebook');
+      await notebook.waitFor();
+      await notebook
+        .getByLabel('Code Cell Content')
+        .getByRole('textbox')
+        .click();
+      await notebook
+        .getByLabel('Code Cell Content')
+        .getByRole('textbox')
+        .fill('%subshell');
+      await page.menu.clickMenuItem('Run>Run Selected Cell');
+
+      const output1 = notebook.locator('.jp-OutputArea-output').locator('pre');
+      const text1 = (await output1.innerText()).split('\n');
+
+      test.skip(
+        text1[0] === 'UsageError: Line magic function `%subshell` not found.',
+        'Test skipped as subshells not supported by kernel'
+      );
+
+      // Subshells supported:
+      expect(text1[0]).toEqual('subshell id: None');
+      expect(text1[5]).toEqual('subshell list: []');
+
+      // Open subshell console
+      await output1.click({ button: 'right' });
+      await page
+        .getByRole('menuitem', { name: 'New Subshell Console for Notebook' })
+        .click();
+
+      // Run %subshell in console
+      const subshellConsole = page.locator('.jp-CodeConsole');
+      await subshellConsole.waitFor();
+      await subshellConsole
+        .getByLabel('Code Cell Content')
+        .getByRole('textbox')
+        .click();
+      await subshellConsole
+        .getByLabel('Code Cell Content')
+        .getByRole('textbox')
+        .fill('%subshell');
+      await page.menu.clickMenuItem('Run>Run Cell (forced)');
+
+      const output2 = subshellConsole
+        .locator('.jp-OutputArea-output')
+        .locator('pre');
+      const text2 = (await output2.innerText()).split('\n');
+      expect(text2[0]).toMatch(/^subshell id:/);
+      const subshellId = text2[0].split(':')[1].trim();
+      expect(text2[5]).toEqual(`subshell list: ['${subshellId}']`);
```

Comment:
Looks like this assertion fails on CI:

![image](https://github.com/user-attachments/assets/e85425dd-df0f-4fb6-a1f2-d4f32c383d74)